tools/documentation_updater/documentation_updater.py

```diff
@@ -0,0 +1,214 @@
+                             '(by default used directory from "--readme-dir" key')
+
+    parser.add_argument('--deprecated_representation', type=bool, default=False,
+                        help="Used for old topology's representation")
+    parser.add_argument('-l', '--list', type=str, default='list_topologies.yml',
+                        help='DEPRECATED: file with topologies list')
+    parser.add_argument('-o', '--out-file', type=str,
+                        help='DEPRECATED: output file with topologies list '
+                             '(by default used original file from --out-file key)')
+
+    parser.add_argument('--regime', type=str, choices=REGIMES, default='check',
+                        help='Script work regime: "check" only finds diffs, "update" - updates values')
+    parser.add_argument('--log-level', choices=LOG_LEVELS.keys(), default='warning',
+                        help='Level of logging')
+    return parser.parse_args()
+
+
+def collect_readme(directory, ignored_files=('index.md',)):
+    if not os.path.isdir(directory):
+        logging.critical("Directory {} does not exist. Please check '--readme-dir' option.".format(os.path.abspath(directory)))
+        exit(1)
+    files = {}
+    for r, d, f in os.walk(directory):
+        for file in f:
+            if '.md' not in file or file in ignored_files:
+                continue
+            files[file.replace(".md", "")] = os.path.join(r, file)
+    logging.info('Collected {} readme files'.format(len(files)))
+    if not files:
+        logging.error("No markdown file found in {}. Exceptions - {}. Ensure, that you set right directory.".format(directory, ignored_files))
+        exit(1)
+    return files
+
+
+def convert(lines):
+    def flatten_links(s):
+        links = re.findall(r"\[.*?\]\(.*?\)", s)
+        for link in links:
+            plain_link = re.sub('[\[\]]', '', link.replace('](', ' (').replace('(', '<').replace(')', '>'))
+            s = s.replace(link, plain_link)
+        return s
+
+    result = ''
+    list_signs = ['-', '*']
+    for line in lines:
+        if len(line.strip()) == 0:
+            result += '\n'
+        elif line.lstrip()[0] in list_signs:
+            result += '\n'
+        if len(result) > 0 and result[len(result)-1] != '\n':
+            result += ' '
+        result += line.rstrip('\n')
+        result = flatten_links(result)
+        result = result.replace("`", "\"").replace("\*", "*")
+    return result.strip()
+
+
+def collect_descriptions(files):
+    descriptions = {}
+    for name, file in files.items():
+        started = False
+        desc = []
+        readme = open(file, "r")
+        for line in readme:
+            if line[0:2] == '##':
+                if not started:
+                    started = True
+                    continue
+                else:
+                    break
+            if started:
+                desc.append(line)
+        desc = convert(desc).strip('\n')
```

Comment:
`convert` already strips all whitespace at the end so this `strip` is redundant.